### PR TITLE
DCKUBE-733: Update the product versions

### PR DESCRIPTION
**Pull request by nanux**
- [x] fix the tests
- [x] add bamboo-agent

### CODE REVIEW

File: src/main/scripts/update_versions.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        logging.info('LTS product (%s)', current_lts[p])
+        version = latest_minor(current_lts[p], mac_versions(p))
+    else:
+        logging.info("Non-LTS product")
+        r = requests.get(f'https://marketplace.atlassian.com/rest/2/products/key/{p}/versions/latest')
+        version = r.json()['name']
+
+    logging.info(f"version: %s, tagname: {version}{suffix}", version)
+    update_chart(p, version)
+    logging.info("updated the product chart")
```
Comment (nanux) [inline src/main/scripts/update_versions.py:74]: Output looks like this:
```
I Updating product versions in helm charts
W Always check the latest available LTS versions on https://confluence.atlassian.com/enterprise/long-term-support-releases-948227420.html
I -------------------------
I Product: bitbucket
I LTS product (7.17)
I version: 7.17.1, tagname: 7.17.1-jdk11
I current version: 7.17.1-jdk11
I updated the product chart
I -------------------------
I Product: jira
I LTS product (8.20)
I version: 8.20.1, tagname: 8.20.1-jdk11
I current version: 8.20.1-jdk11
I updated the product chart
I -------------------------
I Product: bamboo
I Non-LTS product
I version: 8.0.4, tagname: 8.0.4-jdk11
I current version: 8.0.4-jdk11
I updated the product chart
I -------------------------
I Product: confluence
I LTS product (7.13)
I version: 7.13.2, tagname: 7.13.2-jdk11
I current version: 7.13.2-jdk11
I updated the product chart
I -------------------------
I Product: crowd
I Non-LTS product
I version: 4.4.0, tagname: 4.4.0-jdk11
I current version: 4.4.0-jdk11
I updated the product chart
```

File: src/main/scripts/update_versions.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import logging
+
+import requests
```
Comment (nanux) [inline src/main/scripts/update_versions.py:3]: This is just a script to help with the update for the next person doing this. It still needs a bit of handholding (checking the LTS, installing these 2 prerequisites) but I still think it makes it much easier.

File: src/main/scripts/update_versions.py (hunk 1 above)
Comment (tarka) [reply src/main/scripts/update_versions.py:74]: Noice

**Review by tarka — APPROVED** [review]

File: src/main/scripts/update_versions.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+def update_chart(product, new_version):
+    chart_file = f'../charts/{product}/Chart.yaml'
+    with open(chart_file, "r") as stream:
+        content = stream.read()
+        doc = yaml.safe_load(content)
+        current_version = doc['appVersion']
+        logging.info("current version: %s", current_version)
+
+    new_content = content.replace(current_version, f"{new_version}{suffix}")
```
Comment (nanux) [inline src/main/scripts/update_versions.py:21]: Naively I have used `yaml.dump()` first but that just re-shuffle the whole file.

**Review by errcode1202 — APPROVED** [review]

File: src/main/scripts/update_versions.py (hunk 2 above)
Comment (nghazali) [reply src/main/scripts/update_versions.py:3]: We can fetch the LTS product version automatically. There is a python script we have (`./src/test/scripts/product_versions.py`) and we can get use like this:
```
import product_version
...
product = "bamboo"
...
lts_version = product_versions.get_lts_version(product)
...
```

**Pull request merged**